### PR TITLE
feat(select): fires 'dqpl:select:change' when selection is made

### LINKED ISSUE #83: Use JS to get the value of a select / onChange events?
Is there currently a way to use JS to trigger an event when the value changes in a select?
I'm listening to 'blur' events but it triggers even when the value doesn't change.

### LINKED ISSUE #84: Use JS to set the value of a select?
Similar to issue #83 but the inverse...
I'd like to dynamically set the active state of the select.
Is there a way to do this? 
Thanks!

### PR DESCRIPTION
**Pull request by schne324**
closes #83
closes #84

## Example usage

```js
// grab the select button
const select = document.querySelector('[aria-haspopup=listbox]');
select.addEventListener('dqpl:select:change', ({ detail }) => {
  console.log(`New value is ${detail.value}`)
});
```

### CODE REVIEW

File: lib/components/selects/select.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 'use strict';
 
 import Classlist from 'classlist';
+import CustomEvent from 'custom-event';
```
Comment (stephenmathieson) [inline lib/components/selects/select.js]: Just curious, why not use the [`Event` constructor](https://developer.mozilla.org/en-US/docs/Web/API/Event/Event) instead?

Comment (schne324) [reply lib/components/selects/select.js]: I was scared away from using that based on [IE not supporting it at all](https://developer.mozilla.org/en-US/docs/Web/API/Event/Event#Browser_compatibility)

Comment (stephenmathieson) [reply lib/components/selects/select.js]: Ah OK. I assume that IE is supported here then?

Comment (schne324) [reply lib/components/selects/select.js]: 11+ was the (undocumented) original plan. It is a topic that we haven't visited in a while though tbh

Comment (stephenmathieson) [reply lib/components/selects/select.js]: Gotcha, OK. If we do plan/hope to support IE, then this makes sense to me 👍

File: lib/components/selects/select.js [hunk 2]
```diff
@@ -17,6 +18,13 @@ module.exports = (listboxButton, listbox, noHide) => {
   if (active) {
     active.setAttribute('aria-selected', 'true');
     Classlist(active).add('dqpl-option-selected');
+    // fire a custom change event
+    const onChange = new CustomEvent('dqpl:select:change', {
+      detail: {
+        value: active.innerHTML
+      }
```
[the remark this answers is not in the document]
Comment (schne324) [reply lib/components/selects/select.js]: Good question, I too checked the defaults which seemed fine for this case. I figured we'd change it if/when we needed to. I could be wrong here though, do you think we should override the defaults? If so, which ones?

**Comment by stephenmathieson** [comment]
@schne324 I only have read access here, so my approval on this PR does nothing.

**Comment by schne324** [comment]
@stephenmathieson my bad...I just gave our team write access here!

File: lib/components/selects/select.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (AutoSponge) [reply lib/components/selects/select.js]: I think any example should show a listener is capturing since by default this event won't bubble.

**Pull request merged**